tests/fixtures/hdfs.py

```diff
@@ -0,0 +1,195 @@
+import sdk_security
+import shakedown
+import spark_utils as utils
+
+from tests import hdfs_auth
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HDFS_TASK_COUNT = 10
+GENERIC_HDFS_USER_PRINCIPAL = "hdfs@{realm}".format(realm=sdk_auth.REALM)
+ALICE_USER = "alice"
+ALICE_PRINCIPAL = "{user}@{realm}".format(user=ALICE_USER, realm=sdk_auth.REALM)
+KEYTAB_SECRET_PATH = os.getenv("KEYTAB_SECRET_PATH", "__dcos_base64___keytab")
+
+HDFS_PACKAGE_NAME = 'hdfs'
+HDFS_SERVICE_NAME = 'hdfs'
+HISTORY_PACKAGE_NAME = os.getenv("HISTORY_PACKAGE_NAME", "spark-history")
+HISTORY_SERVICE_NAME = os.getenv("HISTORY_SERVICE_NAME", "spark-history")
+
+HDFS_DATA_DIR = '/users/{}'.format(ALICE_USER)
+HDFS_HISTORY_DIR = '/history'
+
+HDFS_KRB5_CONF_ORIG = '''[libdefaults]
+default_realm = %(realm)s
+dns_lookup_realm = true
+dns_lookup_kdc = true
+udp_preference_limit = 1
+
+[realms]
+  %(realm)s = {
+    kdc = kdc.marathon.mesos:2500
+  }
+
+[domain_realm]
+  .hdfs.dcos = %(realm)s
+  hdfs.dcos = %(realm)s
+''' % {"realm": sdk_auth.REALM}  # avoid format() due to problems with "{" in string
+HDFS_KRB5_CONF = base64.b64encode(HDFS_KRB5_CONF_ORIG.encode('utf8')).decode('utf8')
+
+SPARK_SUBMIT_HDFS_KERBEROS_ARGS = ["--kerberos-principal", ALICE_PRINCIPAL,
+                 "--keytab-secret-path", "/{}".format(KEYTAB_SECRET_PATH),
+                 "--conf", "spark.mesos.driverEnv.SPARK_USER={}".format(utils.SPARK_USER)]
+
+HDFS_CLIENT_ID = "hdfsclient"
+SPARK_HISTORY_USER = "nobody"
+
+
+@pytest.fixture(scope='session')
+def configure_security_hdfs():
+    yield from sdk_security.security_session(HDFS_SERVICE_NAME)
+
+
+@pytest.fixture(scope='session')
+def hdfs_with_kerberos(configure_security_hdfs, kerberos_env):
+    try:
+        service_kerberos_options = {
+            "service": {
+                "name": HDFS_SERVICE_NAME,
+                "security": {
+                    "kerberos": {
+                        "enabled": True,
+                        "kdc": {
+                            "hostname": kerberos_env.get_host(),
+                            "port": int(kerberos_env.get_port())
+                        },
+                        "keytab_secret": kerberos_env.get_keytab_path(),
+                        "realm": kerberos_env.get_realm()
+                    }
+                }
+            },
+            "hdfs": {
+                "security_auth_to_local": hdfs_auth.get_principal_to_user_mapping()
+            }
+        }
+
+        sdk_install.uninstall(HDFS_PACKAGE_NAME, HDFS_SERVICE_NAME)
+        sdk_install.install(
+            HDFS_PACKAGE_NAME,
+            HDFS_SERVICE_NAME,
+            DEFAULT_HDFS_TASK_COUNT,
+            additional_options=service_kerberos_options,
+            timeout_seconds=30 * 60)
+
+        yield kerberos_env
+
+    finally:
+        sdk_install.uninstall(HDFS_PACKAGE_NAME, HDFS_SERVICE_NAME)
+        kerberos_env.cleanup()
+
+
+@pytest.fixture(scope='session')
+def setup_hdfs_client(hdfs_with_kerberos):
```

Comment:
It is strange that this depends on `hdfs_with_kerberos` without accessing it. Should the client depend on the service in some way?

Reply:
The main intent of this dependency is to introduce the ordering in which `hdfsclient` never created before hdfs service. What would be a better way of implementing this?

Reply:
I realise that it's about ordering, so that's fine, but the dependence is not really explicit here. It takes knowledge of the client itself to know that it **requires** an HDFS service with a name `hdfs`.

What I have used in the past is something like: https://github.com/mesosphere/dcos-commons/blob/d83139ac0fb9c53719a9e5637a5704ae9dfe23d5/frameworks/hdfs/tests/test_kerberos_auth.py#L101

where the `hdfs_with_kerberos` is a `dict` which contains the required configuration and this si explicitly passed to the client. It's not a requirement for this PR though.